### PR TITLE
Min/max relations

### PR DESCRIPTION
**Pull request by stason314**
Added convenience method for finding max and min between  two relations. Also convenience size configuration with `CGSize`.

### CODE REVIEW

File: Framezilla.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -197,6 +198,13 @@
 			path = Tests;
 			sourceTree = "<group>";
 		};
+		855C4C1B23F57551007AFEEE /* Recovered References */ = {
+			isa = PBXGroup;
+			children = (
+			);
+			name = "Recovered References";
+			sourceTree = "<group>";
+		};
```
Comment (artemnovichkov) [inline Framezilla.xcodeproj/project.pbxproj]: Please, remove it

File: Sources/Maker+Relations.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import UIKit
+
+public extension Maker {
+
+    /// Finds relation with max value
+    ///
+    /// Use this method when you want to find relation with max value.
+    ///
+    /// - parameter lhs:   The left hand side relation to campare.
```
Comment (artemnovichkov) [inline Sources/Maker+Relations.swift]:
```suggestion
    /// - parameter lhs:   The left hand side relation to compare.
```

File: Sources/Maker+Relations.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    /// - parameter rhs:   The right hand side relation to campare.
```
Comment (artemnovichkov) [inline Sources/Maker+Relations.swift]:
```suggestion
    /// - parameter rhs:   The right hand side relation to compare.
```

File: Sources/Maker+Relations.swift [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        let rhsConvertedValue = convertedValue(for: rhs)
+        return lhsConvertedValue > rhsConvertedValue ? lhs : rhs
+    }
+
+    /// Finds relation with min value
+    ///
+    /// Use this method when you want to find relation with min value.
+    ///
+    /// - parameter lhs:   The left hand side relation to campare.
+    /// - parameter rhs:   The right hand side relation to campare.
```
Comment (artemnovichkov) [inline Sources/Maker+Relations.swift]: compare

File: Sources/UIView+Relations.swift [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+/// Phantom type for `nui_left`, `nui_right`, `nui_centerX` relations.
+
+public protocol HorizontalRelation {}
+
+/// Phantom type for `nui_top`, `nui_bottom`, `nui_centerY` relations.
+
+public protocol VerticalRelation {}
+
+/// Phantom type for `nui_height`, `nui_widht` relations.
```
Comment (artemnovichkov) [inline Sources/UIView+Relations.swift:17]:
```suggestion
/// Phantom type for `nui_height`, `nui_width` relations.
```

File: Sources/Maker+Relations.swift [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    ///
+    /// - returns: `RelationView<Relation>` relation with max value between `lhs`and `rhs`.
+
+    func maxRelation<Relation>(_ lhs: RelationView<Relation>, _ rhs: RelationView<Relation>) -> RelationView<Relation> {
+        let lhsConvertedValue = convertedValue(for: lhs)
+        let rhsConvertedValue = convertedValue(for: rhs)
+        return lhsConvertedValue > rhsConvertedValue ? lhs : rhs
```
Comment (artemnovichkov) [inline Sources/Maker+Relations.swift:21]:
```suggestion
        return max(lhsConvertedValue, rhsConvertedValue)
```

File: Sources/Maker+Relations.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    ///
+    /// - parameter lhs:   The left hand side relation to campare.
+    /// - parameter rhs:   The right hand side relation to campare.
+    ///
+    /// - returns: `RelationView<Relation>` relation with min value between `lhs`and `rhs`.
+
+    func minRelation<Relation>(_ lhs: RelationView<Relation>, _ rhs: RelationView<Relation>) -> RelationView<Relation> {
+        let lhsConvertedValue = convertedValue(for: lhs)
+        let rhsConvertedValue = convertedValue(for: rhs)
+        return lhsConvertedValue < rhsConvertedValue ? lhs : rhs
```
Comment (artemnovichkov) [inline Sources/Maker+Relations.swift:36]:
```suggestion
        return min(lhsConvertedValue, rhsConvertedValue)
```

File: Sources/Maker.swift [hunk 8]
```diff
@@ -262,6 +262,16 @@ public final class Maker {
...
 
+    /// Installs constant width and height at the same time.
+    ///
+    /// - parameter size:  The size for view.
+    ///
+    /// - returns: `Maker` instance for chaining relations.
+
+    @discardableResult public func size(_ size: CGSize) -> Maker {
+        return self.size(width: size.width, height: size.height)
+    }
```
Comment (artemnovichkov) [inline Sources/Maker.swift:273]: I don't see any tests for this logic

**Review by artemnovichkov — CHANGES_REQUESTED** [review]

File: Tests/MakerTests.swift [hunk 9]
```diff
@@ -464,6 +464,18 @@ class MakerTests: BaseTest {
...
+
+    func testSizeConfiguration() {
+        let size = CGSize(width: 20, height: 20)
+        testingView.frame = .zero
+        testingView.configureFrame { maker in
+            maker.size(width: size.width, height: size.height)
+            maker.size(size)
+        }
+        XCTAssertEqual(size, testingView.frame.size)
+    }
```
Comment (artemnovichkov) [inline Tests/MakerTests.swift]: Please split this test in two separate tests

**Review by artemnovichkov — CHANGES_REQUESTED** [review]

**Review by artemnovichkov — APPROVED** [review]

File: Sources/Maker+Relations.swift [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    /// Finds relation with max value
+    ///
+    /// Use this method when you want to find relation with max value.
+    ///
+    /// - parameter lhs:   The left hand side relation to compare.
+    /// - parameter rhs:   The right hand side relation to compare.
+    ///
+    /// - returns: `RelationView<Relation>` relation with max value between `lhs`and `rhs`.
+
+    func maxRelation<Relation>(_ lhs: RelationView<Relation>, _ rhs: RelationView<Relation>) -> RelationView<Relation> {
```
Comment (rosberry) [inline Sources/Maker+Relations.swift]:
```suggestion
    func max<Relation>(_ lhs: RelationView<Relation>, _ rhs: RelationView<Relation>) -> RelationView<Relation> {
```

File: Sources/Maker+Relations.swift [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    /// Finds relation with min value
+    ///
+    /// Use this method when you want to find relation with min value.
+    ///
+    /// - parameter lhs:   The left hand side relation to compare.
+    /// - parameter rhs:   The right hand side relation to compare.
+    ///
+    /// - returns: `RelationView<Relation>` relation with min value between `lhs`and `rhs`.
+
+    func minRelation<Relation>(_ lhs: RelationView<Relation>, _ rhs: RelationView<Relation>) -> RelationView<Relation> {
```
Comment (rosberry) [inline Sources/Maker+Relations.swift]:
```suggestion
    func min<Relation>(_ lhs: RelationView<Relation>, _ rhs: RelationView<Relation>) -> RelationView<Relation> {
```

**Review by ghost — APPROVED** [review]

**Review by ntunin — APPROVED** [review]

**Pull request merged**